sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,687 @@
+
+    Representation of a variational inference for a Bayesian Gaussian mixture
+    model probability distribution. This class allows to do inference of an
+    approximate posterior distribution over the parameters of a Gaussian
+    mixture distribution. The number of components can be inferred from the
+    data.
+
+    Read more in the :ref:`User Guide <bgmm>`.
+
+    Parameters
+    ----------
+    n_components: int, defaults to 1.
+        The number of mixture components.
+
+    covariance_type : {'full', 'tied', 'diag', 'spherical'},
+        defaults to 'full'.
+        String describing the type of covariance parameters to use.
+        Must be one of::
+        'full' (each component has its own general covariance matrix).
+        'tied' (all components share the same general covariance matrix),
+        'diag' (each component has its own diagonal covariance matrix),
+        'spherical' (each component has its own single variance),
+
+    tol : float, defaults to 1e-3.
+        The convergence threshold. EM iterations will stop when the
+        lower bound average gain on the likelihood (of the training data with
+        respect to the model) is below this threshold.
+
+    max_iter : int, default to 100.
+        The number of EM iterations to perform.
+
+    n_init : int, default to 1.
+        The number of initializations to perform. The best results is kept.
+
+    init_params : {'kmeans', 'random'}, defaults to 'kmeans'.
+        The method used to initialize the weights, the means and the
+        covariances.
+        Must be one of::
+        'kmeans' : responsibilities are initialized using kmeans.
+        'random' : responsibilities are initialized randomly.
+
+    dirichlet_concentration_prior : float, optional.
+        The user-provided dirichlet concentration prior parameter of the
+        Dirichlet distribution. The higher concentration puts more mass in the
+        center and will lead to more components being active, while a lower
+        concentration parameter will lead to more mass at the edge of the
+        simplex. The value of the parameter must be greater than 0. If is None,
+        the dirichlet concentration prior is set to `1. / n_components`.
+
+    mean_precision_prior : float, optional.
+        The user-provided mean precision prior parameter of the Gaussian
+        distribution. Controls the extend to where means can be placed. Smaller
+        values concentrates the means of each clusters around `mean_prior`.
+        The value of the parameter must be greater than 0.
+        If it is None, mean precision prior is set to 1.
+
+    mean_prior : array-like, shape (`n_features`,), optional
+        The user-provided mean prior of the Gaussian distribution.
+        If it is None, the mean prior is set to the mean of X.
+
+    freedom_degrees_prior : float, optional.
+        The user-provided number of degrees of freedom prior parameter of the
```

Comment:
Rephrase.

The prior on the number of degrees of freedom for the covariance distribution.